crates/astria-sequencer-types/src/sequencer_block_data.rs

```diff
@@ -63,15 +72,31 @@ impl SequencerBlockData {
         header: Header,
         last_commit: Option<Commit>,
         rollup_data: HashMap<Namespace, RollupData>,
+        action_tree_root: Hash,
+        action_tree_root_inclusion_proof: InclusionProof,
     ) -> eyre::Result<Self> {
+        // perform data validations to ensure only valid [`SequencerBlockData`]
+        // can be constructed
+        let Some(data_hash) = header.data_hash else {
+            bail!(Error::MissingDataHash);
+        };
+        action_tree_root_inclusion_proof
+            .verify(action_tree_root.as_bytes(), data_hash)
+            .wrap_err("failed to verify action tree root inclusion proof")?;
+        // TODO: also verify last_commit_hash (#270)
```

Comment:
nit: full link preferable

Reply:
Please provide a full link to the issue or PR, like so:
```suggestion
        // TODO(https://github.com/astriaorg/astria/issues/270): also verify last_commit_hash
```